src/components/project/list-filter.query.ts

```diff
@@ -38,6 +43,27 @@ export const projectListFilter = (input: ProjectListInput) =>
       relation('out', '', 'partner', ACTIVE),
       node('', 'Partner', { id }),
     ]),
+    userId:
+      ({ query, value }) =>
+      () =>
+        value
+          ? query.where([
+              path([
+                node('node'),
+                relation('out', '', 'member', ACTIVE),
+                node('', 'ProjectMember'),
+                relation('out', '', 'user', ACTIVE),
+                node('', 'User', { id: value }),
+              ]),
+              path([
+                node('node'),
+                relation('out', '', 'engagement', ACTIVE),
+                node('', 'Engagement'),
+                relation('out', '', 'intern', ACTIVE),
+                node('', 'User', { id: value }),
+              ]),
+            ])
+          : null,
```

Comment:
You can re-use the single where clause the builder builds. The function won't be called if the value is nullable, so that check isn't necessary either.
```ts
({ value }) => ({
  userId: [
    path([...]),
    path([...]),          
  ],
})
```